backend/src/apiserver/resource/resource_manager.go

```diff
@@ -544,6 +550,70 @@ func (r *ResourceManager) RetryRun(runId string) error {
 	return nil
 }
 
+func (r *ResourceManager) ReadLog(runId string, nodeId string, dst io.Writer) error {
+	run, err := r.checkRunExist(runId)
+	if err != nil {
+		return util.NewBadRequestError(errors.New("log cannot be read"), "Run does not exist")
+	}
+
+	err = r.readRunLogFromPod(run, nodeId, dst)
+	if err != nil {
```

Comment:
For this error, can we log it with info or warning? On my env, I was having a rbac permission error, but I couldn't see it from the logs since it get ignored/overwritten over here.

Reply:
Error handling is enhanced now